cpp/src/io/comp/nvcomp_adapter.cpp

```diff
@@ -321,40 +343,64 @@ static void batched_compress_async(compression_type compression,
   CUDF_EXPECTS(nvcomp_status == nvcompStatus_t::nvcompSuccess, "Error in compression");
 }
 
-bool is_aligned(void const* ptr, std::uintptr_t alignment) noexcept
+constexpr bool is_aligned(void const* ptr, std::uintptr_t alignment) noexcept
 {
   return (reinterpret_cast<std::uintptr_t>(ptr) % alignment) == 0;
 }
 
+// since some compressors have extreme temp memory requirements, scale back the number
+// of chunks to process until the request for temp memory succeeds.
+std::pair<rmm::device_buffer, size_t> compress_temp_buffer(compression_type compression,
```

Comment:
This will be fun to implement with GetTempSizeEx (takes total uncompressed size instead of max chunk size).
Looks good for the current GetTempSize API.